src/urllib3/connection.py

```diff
@@ -199,6 +199,12 @@ def request(self, method, url, body=None, headers=None):
         else:
             # Avoid modifying the headers passed into .request()
             headers = headers.copy()
+            # Don't send bytes keys to httplib to avoid bytes/str comparison
+            for key, value in list(headers.items()):
+                if isinstance(key, bytes):
+                    headers.pop(key)
+                    headers[key.decode()] = value
```

Comment:
Should we be explicit here and use `key.decode("ascii")`?

Reply:
I would agree it should be explicit

Reply:
It's not only about being explicit: the default `decode()` parameter is `utf-8`. While `ascii` is the right thing to do, httplib currently decodes to `latin-1`, so this is what I changed this line to, in order to avoid a breaking change. (`latin-1` is also superset of ASCII, and has the "nice" property of not raising an error if the true encoding is something else like UTF-8.)